sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -281,6 +281,34 @@ export class ManagementClient extends LinkEntity {
       : undefined;
   }
 
+  /**
+   * Evaluates the `associatedLinkName` to use for given Service Bus entity client instance
+   * @param forSender Boolean flag indicating if this for an operation on sender or receiver
+   * @param clientEntityContext The `ClientEntityContext` associated with given Service Bus entity client
+   * @param sessionId `sessionId` if applicable
+   */
+  private _getAssociatedLinkName(
+    forSender: boolean,
+    clientEntityContext: ClientEntityContext,
+    sessionId?: string
+  ): string | undefined {
+    let associatedLinkName: string | undefined;
+    if (forSender && clientEntityContext.sender) {
+      return clientEntityContext.sender.name;
+    } else {
+      if (sessionId != undefined) {
+        if (clientEntityContext.messageSessions[sessionId]) {
+          associatedLinkName = clientEntityContext.messageSessions[sessionId].name;
+        }
+      } else if (clientEntityContext.batchingReceiver) {
+        associatedLinkName = clientEntityContext.batchingReceiver.name;
+      } else if (clientEntityContext.streamingReceiver) {
+        associatedLinkName = clientEntityContext.streamingReceiver.name;
+      }
+      return associatedLinkName;
```

Comment:
Instead of setting the variable, you can directly return it like it is done above.

Reply:
I agree. And the return model also lets us skip the multiple nesting